### PR TITLE
tidb-in-kubernetes: add default cluster requirements

### PR DESCRIPTION
**Pull request by tennix**
### What is changed, added or deleted? 

* Add default cluster requirements: `storageclass` and deployment topology. These two aspects usually lead to deployment failures if users are unaware of it.
* Add note about multiple clusters management.

### What is the related PR or file link(s)? 

The storage class part is related to https://github.com/pingcap/docs-cn/pull/1770

### Which version does your change affect? 

dev and v3.0

### CODE REVIEW

File: dev/tidb-in-kubernetes/deploy/general-kubernetes.md [hunk 1]
```diff
@@ -30,7 +30,13 @@ helm inspect values pingcap/tidb-cluster --version=<chart-version> > /home/tidb/
 > - `chart-version` 是 tidb-cluster chart 发布的版本，可以通过 `helm search -l tidb-cluster` 查看当前支持的版本。
 > - 下文会用 `values.yaml` 指代 `/home/tidb/<release-name>/values-<release-name>.yaml`。
 
-根据需要修改上述配置文件，有关配置信息请参考 [TiDB 集群部署配置文档](/tidb-in-kubernetes/reference/configuration/tidb-cluster.md)。
+默认集群使用 `local-storage` 存储类型，生产环境推荐使用本地存储，但实际 Kubernetes 集群中本地存储可能按磁盘类型进行了分类，例如 `nvme-disks`, `sas-disks`，如果是 demo 环境或功能性验证，可以使用网络存储，例如 `ebs`, `nfs` 等，另外 TiDB 集群不同组件对磁盘的要求不一样。所以部署集群前要根据当前 Kubernetes 集群支持的存储类型以及使用场景为 TiDB 集群各组件选择合适的存储类型，通过修改 `values.yaml` 中各组件的 `storageClassName` 字段设置存储类型。关于 Kubernetes 集群支持哪些存储类型，请联系系统管理员确定。
+
+如果创建集群时设置了集群中不存在的存储类型，则会导致集群创建处于 Pending 状态，需要将[集群彻底销毁掉](/tidb-in-kubernetes/maintain/destroy-tidb-cluster.md)。
+
+默认部署的集群拓扑是：3 个 PD，3 个 TiKV，2 个 TiDB 和 1 个监控。TiDB Operator 扩展调度器根据数据高可用要求 Kubernetes 集群至少有 3 个节点，否则请减小默认部署的 PD 和 TiKV 个数为 1，或者将 `values.yaml` 中 `schedulerName` 改为 Kubernetes 内置调度器 `default-scheduler`。
```
[the remark this answers is not in the document]
Comment (tennix) [reply dev/tidb-in-kubernetes/deploy/general-kubernetes.md]: 我再加些强调说明改成 `default-scheduler` 后的影响吧，这种一般是 demo 环境。

File: dev/tidb-in-kubernetes/deploy/general-kubernetes.md [hunk 2]
```diff
@@ -30,7 +30,17 @@ helm inspect values pingcap/tidb-cluster --version=<chart-version> > /home/tidb/
 > - `chart-version` 是 tidb-cluster chart 发布的版本，可以通过 `helm search -l tidb-cluster` 查看当前支持的版本。
 > - 下文会用 `values.yaml` 指代 `/home/tidb/<release-name>/values-<release-name>.yaml`。
 
-根据需要修改上述配置文件，有关配置信息请参考 [TiDB 集群部署配置文档](/tidb-in-kubernetes/reference/configuration/tidb-cluster.md)。
+默认集群使用 `local-storage` 存储类型。生产环境推荐使用本地存储，但实际 Kubernetes 集群中本地存储可能按磁盘类型进行了分类，例如 `nvme-disks`，`sas-disks`。如果是演示环境或功能性验证，可以使用网络存储，例如 `ebs`，`nfs` 等。另外 TiDB 集群不同组件对磁盘的要求不一样。所以部署集群前要根据当前 Kubernetes 集群支持的存储类型以及使用场景为 TiDB 集群各组件选择合适的存储类型，通过修改 `values.yaml` 中各组件的 `storageClassName` 字段设置存储类型。关于 Kubernetes 集群支持哪些[存储类型](/tidb-in-kubernetes/reference/configuration/storage-class.md)，请联系系统管理员确定。
+
+如果创建集群时设置了集群中不存在的存储类型，则会导致集群创建处于 Pending 状态，需要将[集群彻底销毁掉](/tidb-in-kubernetes/maintain/destroy-tidb-cluster.md)。
```
Comment (lilin90) [inline dev/tidb-in-kubernetes/deploy/general-kubernetes.md]:
```suggestion
如果创建集群时设置了集群中不存在的存储类型，则会导致集群创建处于 Pending 状态，需要[将集群彻底销毁掉](/tidb-in-kubernetes/maintain/destroy-tidb-cluster.md)。
```

File: dev/tidb-in-kubernetes/deploy/general-kubernetes.md [hunk 3]
```diff
@@ -30,7 +30,17 @@ helm inspect values pingcap/tidb-cluster --version=<chart-version> > /home/tidb/
...
+
+默认部署的集群拓扑是：3 个 PD，3 个 TiKV，2 个 TiDB 和 1 个监控。根据数据高可用原则，TiDB Operator 扩展调度器要求 PD 集群和 TiKV 集群至少有 3 个节点，否则，请减小默认部署的 PD 和 TiKV 个数为 1，或者将 `values.yaml` 中 `schedulerName` 改为 Kubernetes 内置调度器 `default-scheduler`。
```
Comment (lilin90) [inline dev/tidb-in-kubernetes/deploy/general-kubernetes.md]: - `3 个 PD` means "instance" or "node"? We need to keep the wording consistent.
- This sentence is confusing: `否则，请减小默认部署的 PD 和 TiKV 个数为 1，`. Could you make the logic clearer?

File: dev/tidb-in-kubernetes/deploy/general-kubernetes.md [hunk 4]
```diff
@@ -49,3 +59,5 @@ helm install pingcap/tidb-cluster --name=<release-name> --namespace=<namespace>
 ``` shell
 kubectl get po -n <namespace> -l app.kubernetes.io/instance=<release-name>
 ```
+
+单个 Kubernetes 集群中可以利用 TiDB Operator 部署管理多套 TiDB 集群，重复上面命令并将 `release-name` 替换成不同名字即可。不同集群可以在相同 `namespace` 中也可以是在不同 `namespace` 中，可根据实际需求选择。
```
Comment (lilin90) [inline dev/tidb-in-kubernetes/deploy/general-kubernetes.md]:
```suggestion
单个 Kubernetes 集群中可以利用 TiDB Operator 部署管理多套 TiDB 集群，重复以上命令并将 `release-name` 替换成不同名字即可。不同集群既可以在相同 `namespace` 中，也可以在不同 `namespace` 中，可根据实际需求进行选择。
```

File: v3.0/tidb-in-kubernetes/deploy/general-kubernetes.md [hunk 5]
```diff
@@ -50,3 +60,5 @@ helm install pingcap/tidb-cluster --name=<release-name> --namespace=<namespace>
 ``` shell
 kubectl get po -n <namespace> -l app.kubernetes.io/instance=<release-name>
 ```
+
+单个 Kubernetes 集群中可以利用 TiDB Operator 部署管理多套 TiDB 集群，重复上面命令并将 `release-name` 替换成不同名字即可。不同集群可以在相同 `namespace` 中也可以是在不同 `namespace` 中，可根据实际需求选择。
```
Comment (lilin90) [inline v3.0/tidb-in-kubernetes/deploy/general-kubernetes.md]:
```suggestion
单个 Kubernetes 集群中可以利用 TiDB Operator 部署管理多套 TiDB 集群，重复以上命令并将 `release-name` 替换成不同名字即可。不同集群既可以在相同 `namespace` 中，也可以在不同 `namespace` 中，可根据实际需求进行选择。
```

File: dev/tidb-in-kubernetes/deploy/general-kubernetes.md (hunk 3 above)
Comment (tennix) [reply dev/tidb-in-kubernetes/deploy/general-kubernetes.md]: 这里的个数是指各个组件部署的个数，云上实例 (instance) 一般是指一个对外提供的服务单元，TiDB 集群是作为一个整体对外提供服务的，是一个最小单元，所以一个 TiDB 实例是指一个 TiDB 集群。而像这里 PD, TiKV, TiDB 个数在云上一般叫节点个数，这里节点与机器不是一对一的关系。

Comment (lilin90) [reply dev/tidb-in-kubernetes/deploy/general-kubernetes.md]: @tennix Got it. Thanks!

Comment (lilin90) [reply dev/tidb-in-kubernetes/deploy/general-kubernetes.md]: @juliezhang1112 FYI.

**Pull request merged**